### ISSUE TITLE
Can't see ERH in the Type Menu

### ISSUE THREAD

**Issue by pontifex-tech** [vnh1968/ERH#2]
Hello, I hope I did everything as written in Wiki but I can't see "Extended Running Headers & Page Numbering" there. I switched Id to en_US but there is no change. I tried also to move the folder inside Scipts Master: ERH1335 inside and outside ERH-master folder. Nothing changed.
I have been using the latest version of Indesing.

![indesign-erh-hidden](https://user-images.githubusercontent.com/65412641/82074823-a183e000-96db-11ea-811d-9d69066b0040.png)

What could you advise me at this point?

**Comment by aarontheman** [comment]
Same issue for me. On Mac OS 10.14, InDesign 2020.

**Comment by vnh1968** [comment]
The script file must be placed in a folder called "startup scripts," which is located in the "scripts" folder. Then you should restart the ID.

**Comment by vnh1968** [comment]
Some more information. Do not try to run the script from the script panel, it is launched from the menu if your publication contains master spreads with running headers.

**Comment by aarontheman** [comment]
Thanks @vnh1968. I had already followed both those instructions, per the Wiki, and my doc does use master spread with text variable running head, but it doesn't work...

**Comment by vnh1968** [comment]
Please, make a package and send it to vnh1968@gmail.com


ср, 10 черв. 2020 о 01:22 aarontheman <notifications@github.com> пише:

**Comment by vnh1968** [comment]
[ https://www.marspremedia.com/software/how-to-adobe-cc ]
There are two different paths on Mac:

Macintosh HD:Applications:Adobe InDesign 2020:Scripts:startup scripts
Macintosh HD:Users:*UserName*:Library:Preferences:Adobe InDesign:*Version
15.0:en_US*:Scripts:startup scripts

perhaps only one of them works. Now I have no way to run the script on the mac,
but the script is in no way tied to system resources. The latest version starts
even on localized versions of ID (russian, ukrainian, and polish). But
this only
applies to 'windows'.

(Variable chunks of the 2nd path underlined   or between asterisks)


ср, 10 черв. 2020 о 01:22 aarontheman <notifications@github.com> пише:

**Comment by aarontheman** [comment]
@vnh1968 Thanks; I tried adding a 'startup scripts' inside the Scripts folder of the User:Library path as you specified, and added your script to it. But still the option doesn't appear in the Type menu. Happy to send a package to you but I'm not sure what package you're referring to...?

**Comment by vnh1968** [comment]
Let's check our actions once again: the ZIP archive is downloaded from
Github, inside it there is
the 'ERH-Master' folder, and finally, there are 2 files in it:
'README.md' and 'erh2021_2.jsx'.
You need to extract the file 'erh2021_2.jsx' and place it in the
folder 'startup scripts'. After
that restart InDesign. If this does not work, let's try to run the
script in the previous version ID mode:

Create folder named 'Version 14.0 Scripts'
inside the 'Scripts' folder (One need to try again two options for the
path to the 'Scripts' folder, as last time.), then create folder 'startup
scripts' in it, and put there the script file 'erh2021_1.jsx' (not ZIP
file or folder ERH-Master).

Let me know if it works for you.

Now I use Windows at work and at home, I used to work with ID on MacOS
and Windows at the same time a few
years ago.
The development and execution of scripts did not cause any special
problems, which gave confidence
in their compatibility for different operating systems. Apparently, I
will have to look again for
the ability to test scripts separately for MacOS.

If a problem arises again, maybe we’ll try to contact using TeamViewer
application, it would allow me at least to see what is happening.

чт, 11 черв. 2020 о 09:23 aarontheman <notifications@github.com> пише:

**Comment by vnh1968** [comment]
As for the package:

a user of the script from Poland created very complex publication files,
I showed him how to work with this script using only one master spread
instead of
more than 10 (according to the number of chapters in the book).

We also solved the problem of running the script in a localized
version of InDesign.
It turns out there are differences for such versions.





чт, 11 черв. 2020 о 09:23 aarontheman <notifications@github.com> пише:

**Comment by aarontheman** [comment]
@vnh1968 Thanks for your continued efforts! I tried that but still, 'no dice'. Shown here is the script installed in the application Scripts folder. (Before that, I tried in the user/Library folder...)

![image](https://user-images.githubusercontent.com/5438892/84427562-f4b55980-abd9-11ea-8b53-a20944fde021.png)

![image](https://user-images.githubusercontent.com/5438892/84427633-16164580-abda-11ea-99f2-2be23438f50d.png)

**Comment by vnh1968** [comment]
I will consult for possible reasons.
Perhaps I will send you test small scripts to understand the problem,
because nobody described similar behavior.


чт, 11 черв. 2020 о 21:52 aarontheman <notifications@github.com> пише:

**Comment by vnh1968** [comment]
Version 2021.3.  
Need to check it out on MacOS.
Some changes to control the script through the menu. 
A menu item (ERH Rebuild) has been added to menu **Type** (to which you can assign a shortcut to your preference) to initialize the rebuilding of running headers after adding or removing pages.

At first I wanted to do an automatic rebuild by tracking changes in the number of pages in the document, but so far there are problems. For large documents, more time is needed for a complete rebuild, and this can interfere with the user's work. Therefore, the restructuring should not be complete, but only in that part of the document that has undergone a change.

Also in the archive you can find the InDesign document on which the script was finalized.

**Comment by vnh1968** [comment]
Hi aarontheman!

I updated a bit of the script code, especially in the run part.
Still, I would like it to run on MacOS.
The launch problem that occurred on Windows 10 was due to the fact
that the version of InDesign was with non-English localization.
After some changes in the code, everything started working in any localization.

Unzip the script into the folder 'startup scripts'
(where the file "ForceDirectory.txt" is located), the folder
'Version 14 scripts' is not needed.
Reload ID. I really hope that everything will work.


чт, 11 черв. 2020 о 21:52 aarontheman <notifications@github.com> пише:
